sdk/src/main/java/com/mapbox/maps/renderer/widget/Widget.kt

```diff
@@ -10,12 +10,44 @@ abstract class Widget internal constructor() {
   internal abstract val renderer: WidgetRenderer
 
   /**
-   * Set absolute translation of widget in pixels.
+   * Update the widget to the new position.
    */
-  abstract fun setTranslation(translateX: Float, translateY: Float)
+  abstract fun setPosition(widgetPosition: WidgetPosition)
 
   /**
-   * Set absolute rotation of widget in angles.
+   * Get the current position of the widget.
+   */
+  abstract fun getPosition(): WidgetPosition
+
+  /**
+   * Set the translation of the widget in pixels, relative to it's current position.
+   *
+   * @param translateX the offset in pixels towards the right of the screen.
+   * @param translateY the offset in pixels towards the bottom of the screen.
+   */
+  @Deprecated(
+    message = "setTranslation is deprecated, please use setPosition instead.",
+    replaceWith = ReplaceWith("setPosition")
+  )
+  fun setTranslation(translateX: Float, translateY: Float) {
+    val currentPosition = getPosition()
+    setPosition(
+      WidgetPosition {
+        horizontalAlignment = currentPosition.horizontalAlignment
+        verticalAlignment = currentPosition.verticalAlignment
+        offsetX = currentPosition.offsetX + translateX
```

Comment:
do we need to do sign change here also?

Reply:
I don't think so, from the previous logic [here](https://github.com/mapbox/mapbox-maps-android/blob/e05cb7c86ffe47eb1be00319c0e0266be6fc4f0a/sdk/src/main/java/com/mapbox/maps/renderer/widget/BitmapWidgetRenderer.kt#L270), the translationX is directly applied on top of `leftX()`, regardless of the alignment. I think this is also the confusing/inconsistent part of the previous API surface.

Reply:
Thanks, now I see.